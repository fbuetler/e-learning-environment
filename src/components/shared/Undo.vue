<template>
  <div
    id="undo"
    class="flex-item flex-center card clickable responsive"
    title="Neu starten!"
    @click="$emit('undo-operation', $event)"
  >
    <img :src="require('@/assets/icons/undo.png')" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";

@Component<Undo>({})
export default class Undo extends Vue {}
</script>

<style scoped>
.card > img {
  margin: 1em;
}
</style>
