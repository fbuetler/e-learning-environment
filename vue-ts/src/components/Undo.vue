<template>
  <div
    id="undo"
    class="undo flex-item flex-center card"
    @click="$emit('undo-operation', $event)"
    title="Neu starten!"
  >
    <img :src="require('@/assets/icons/undo.png')" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";

@Component<Undo>({})
export default class Undo extends Vue {}
</script>

<style scoped>
.undo > img {
  width: 100%;
  max-width: 80px;
  margin: 1em;
}
</style>
