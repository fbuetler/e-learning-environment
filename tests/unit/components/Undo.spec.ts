import { shallowMount } from "@vue/test-utils";
import Undo from "@/components/shared/Undo.vue";

describe("Undo.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Undo);
  });

  it("is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("event triggered", async () => {
    await wrapper.find("#undo").trigger("click");
    expect(wrapper.emitted("undo-operation")).toBeTruthy();
  });
});
