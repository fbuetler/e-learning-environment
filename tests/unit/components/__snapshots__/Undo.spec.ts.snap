// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Undo.vue renders correctly 1`] = `
<div
  class="undo flex-item flex-center card clickable"
  id="undo"
  title="Neu starten!"
>
  <img
    src=""
  />
</div>
`;
